### ISSUE TITLE
rose_bunch: add per-instance environment variables

### ISSUE THREAD

**Issue by benfitzpatrick** [metomi/rose#1954]
A common pattern is to have different environment variables set per `rose_bunch` instance.

At the moment, one way to do this is to set them via `env` like this:

```
mode=rose_bunch

[bunch]
command-format=env INPUT_PATH=$ROSE_DATA/%(SUNGLASSES)s wear_my_sunglasses

[bunch-args]
SUNGLASSES=aviator wraparound teashades
```

It might be nicer if we had a separate environment variable specification area - e.g. if we could write:

```
mode=rose_bunch

[bunch]
command-format=wear_my_sunglasses

[bunch-env]
INPUT_PATH=$ROSE_DATA/%(SUNGLASSES)s

[bunch-args]
SUNGLASSES=aviator wraparound teashades
```

**Comment by arjclark** [comment]
Your former example is pretty much the way I'd prefer to do this.

For consistency (assuming I'm reading what your example is trying to do) I'd have written your second example as:
```
...
[bunch-env]
INPUT_PATH=$ROSE_DATA/aviator $ROSE_DATA/wraparound $ROSE_DATA/teashades
...
```
since all the other arglists have to be consistent in length.

Additionally, I'm not particularly mad keen on setting up what are essentially dynamic environment variables in what is a configuration file when the existing system is capable of doing it as part of the command.

Note also, the intended use of rose bunch for your type of problem would be to have a wrapper script along the lines of:
```
#!/bin/bash
export INPUT_PATH=$ROSE_DATA/$1

wear_my_sunglasses
```
with an app config of:
```
mode=rose_bunch

[bunch]
command-format=sunglasses_wrapper %(SUNGLASSES)s

[bunch-args]
SUNGLASSES=aviator wraparound teashades
```

This is not massively dissimilar to what we do for the um-recon, the original use case for the creation of the rose_bunch built-in app.

**Comment by arjclark** [comment]
Related request from #1919 is to have a bunch instance specific environment variable along lines of `ROSE_BUNCH_LOG_ID` or the bunch task's name/number.

**Comment by arjclark** [comment]
`ROSE_BUNCH_LOG_ID` part of this addressed by #2011. Other requirements should be addressed as discussed above.
